### PR TITLE
MWPW-156135 - [editorial-card] bugs

### PR DESCRIPTION
**Pull request by ryanmparrish**
Fixed the following bugs
1) In Editorial card, when image is added per breakpoint, it uses the same decorate bg function with rounded-corder.css where the image unintentionally gets targeted with the class `.background` causing a style naming collision. I am passing this className as a parameter now. 
2) When per-viewport bg images are used, they do not display
3) Button and product lockup override work ONLY ATM
4) Custom device support text should always be body-xxs
5) A [text-block] background doesn't show when it's authored in a wrapping section that also has a background. *bonus

Resolves: [MWPW-156135](https://jira.corp.adobe.com/browse/MWPW-156135)

**Test URLs:**
Before: https://main--milo--adobecom.hlx.page/drafts/rparrish/cards/editorial-cards-bugz?martech=off
After: https://rparrish-editorial-bugz--milo--adobecom.hlx.page/drafts/rparrish/cards/editorial-cards-bugz?martech=off

### CODE REVIEW

File: libs/blocks/editorial-card/editorial-card.css [hunk 1]
```diff
@@ -248,3 +265,25 @@
...
+
+/* Tablet up */
+@media screen and (min-width: 600px) {
+  .editorial-card .vp-media .mobile-only,
+  .editorial-card .vp-media .desktop-only {
+    display: none;
+  }
+
+  .editorial-card .vp-media .tablet-only { display: block; }
+
```
[the remark this answers is not in the document]
Comment (ryanmparrish) [reply libs/blocks/editorial-card/editorial-card.css]: Thanks @overmyheadandbody, I fixed those. Looking into my style-lint settings..

File: libs/blocks/text/text.css [hunk 2]
```diff
@@ -31,7 +31,7 @@
   position: absolute;
   right: 0;
   top: 0;
-  z-index: -1;
+  z-index: 0;
```
[the remark this answers is not in the document]
Comment (ryanmparrish) [reply libs/blocks/text/text.css]: You're right, This doesn't need to be set at all. I noticed this issue when regression testing where the text bg can go behind a section bg. I've added an example in my draft now and mention in the desc.

**Review by robert-bogos — APPROVED** [review]
Looks good 👍

**Review by overmyheadandbody — APPROVED** [review]

**Review by mokimo — APPROVED** [review]

**Review by NadiiaSokolova — APPROVED** [review]
Verified. Ready for Stage.
Testing details https://jira.corp.adobe.com/browse/MWPW-156135

File: libs/blocks/editorial-card/editorial-card.css [hunk 3]
```diff
@@ -248,3 +263,24 @@
 
 .editorial-card.footer-align-left .action-area { justify-content: start; }
 .editorial-card.footer-align-center .action-area { justify-content: center; }
+
+
+/* Tablet up */
+@media screen and (min-width: 600px) {
+  .editorial-card .vp-media .mobile-only,
+  .editorial-card .vp-media .desktop-only {
```
Comment (narcis-radu) [inline libs/blocks/editorial-card/editorial-card.css:271]: I like the consistency, but I think this line is not really needed, `desktop-only` is already hidden on mobile.
Screenshot 2024-09-04 at 21 03 34.

We can keep the code for consistency purposes however.

**Review by narcis-radu — APPROVED** [review]

**Pull request merged**